insights/tests/client/connection/test_test_connection.py

```diff
@@ -0,0 +1,76 @@
+from logging import ERROR
+from unittest.mock import patch
+
+from pytest import fixture
+from pytest import mark
+from requests.exceptions import ConnectTimeout
+from requests.exceptions import ReadTimeout
+from requests.exceptions import SSLError
+
+from insights.client.config import InsightsConfig
+from insights.client.connection import InsightsConnection
+
+EXCEPTION_MESSAGE = "exception"
+EXPECTED_EXCEPTIONS = [SSLError, ConnectTimeout, ReadTimeout]
+
+LOGGING_FILE = "logging-file.log"
+
+parametrize_exceptions = mark.parametrize(
+    ["exception"],
+    [(exception(EXCEPTION_MESSAGE),) for exception in EXPECTED_EXCEPTIONS],
+)
+
+
+@fixture
+@patch("insights.client.connection.InsightsConnection._init_session")
+@patch("insights.client.connection.InsightsConnection.get_proxies")
+def insights_connection(get_proxies, init_session):
+    config = InsightsConfig(base_url="www.example.com", logging_file=LOGGING_FILE)
+
+    connection = InsightsConnection(config)
+    connection.proxies = None
+
+    return connection
+
+
+@parametrize_exceptions
+@patch("insights.client.connection.InsightsConnection._test_urls")
+def test_request_failed_test_urls(test_urls, exception, insights_connection):
```

Comment:
Could you add docstrings to these methods, to shortly describe what they are testing in real-world scenarios?

E.g.
```py
"""Stop the whole connection test when an API call fails."""
```

Reply:
Sure thing!